### PR TITLE
Add solution for singleNumber problem

### LINKED ISSUE #5: Do Challenge 136
https://leetcode.com/problems/single-number/

### PR DESCRIPTION
**Pull request by ChrisLGardner**
There's probably a neater solution than this but it seemed like the easiest way to track which numbers had already been seen:

Runtime: 24 ms, faster than 31.93% of Go online submissions for Single Number.
Memory Usage: 6.3 MB, less than 13.87% of Go online submissions for Single Number.

### CODE REVIEW

**Comment by thomasrayner** [comment]
That's an interesting way to keep track of it. Given your results, I'm guessing there's a lib in Go that functions like Linq that would have basically done this one for you?

**Comment by ChrisLGardner** [comment]
> 
> 
> That's an interesting way to keep track of it. Given your results, I'm guessing there's a lib in Go that functions like Linq that would have basically done this one for you?

Not in the standard library that I've seen but there might be something out there that someone wrote to do this sort of thing. Some sort of grouping function would be nice but is reasonably easy to write like this.

File: chris/single-number.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package chris
+
+func singleNumber(nums []int) int {
+
+	if len(nums) == 1 {
+		return nums[0]
+	}
+
+	result := map[int]int{}
+	for _, v := range nums {
```
Comment (thomasrayner) [inline chris/single-number.go:10]: In Go, how does this syntax work? Is it basically `for "Index in the collection", "value itself" in "collection"`?

Comment (ChrisLGardner) [reply chris/single-number.go:10]: Basically yes, and you use _ to ignore that value. Range will give me both and I don't actually care about the index so I just ignore that.

Comment (thomasrayner) [reply chris/single-number.go:10]: Neat. I haven't had a chance to goof around with Go very much, so this is a neat way to start looking at it - comparing solutions to a common problem.

Comment (ChrisLGardner) [reply chris/single-number.go:10]: Yeah, I need to look at your TS solution to get a better idea of how that works too.

**Comment by glennsarti** [comment]
Nice ... Our Go solutions are basically the same!

**Pull request merged**